physics/cu_unified_deep.F90

```diff
@@ -0,0 +1,6052 @@
+              exit
+          endif
+        enddo
+       endif
+       if(kfinalzu.le.kbcon(i)+1) ierr(i)=41
+       !~ print*,'hco5=',k,kfinalzu,ktop(i),kbcon(i)+1,ierr(i);call flush(6)
+      !
+      endif
+     enddo
+!$acc end parallel
+  end subroutine get_cloud_top
+
+  subroutine calculate_updraft_velocity(its,itf,ktf,ite,kts,kte,ierr,progsigma,   &
+       k22,kbcon,ktcon,zo,entr_rate_2d,cd,fv,rd,el2orc,qeso,to,qo,po,dbyo,    &
+       clw_all,qlk,delp,zu,wu2,omega_u,zeta,wc,omegac,zdqca)
+
+    implicit none
+    logical, intent(in) :: progsigma
+    integer, intent(in) :: itf,its,ktf,ite,kts,kte
+    integer, dimension (its:ite), intent(inout) :: ierr
+    real(kind=kind_phys), dimension (its:ite,kts:kte),intent (in) :: zo,entr_rate_2d,   &
+         cd,po,qeso,to,qo,dbyo,clw_all,qlk,delp,zu
+    integer, dimension (its:ite),intent(in) :: k22,kbcon,ktcon
+    real(kind=kind_phys), dimension (its:ite) :: sumx
+    real(kind=kind_phys) ,intent (in) :: fv,rd,el2orc 
+    real(kind=kind_phys), dimension (its:ite,kts:kte) :: drag, buo, zi, del
+    real(kind=kind_phys), dimension (its:ite,kts:kte),intent (out) :: wu2,omega_u,      &
+         zeta,zdqca
+    real(kind=kind_phys), dimension (its:ite),intent(out) :: wc,omegac
+    real(kind=kind_phys) :: rho,bb1,bb2,dz,dp,ptem,tem1,ptem1,tem,rfact,gamma,val
+    integer :: i,k    
+
+  
+    !  compute updraft velocity square(wu2)
+    !> - Calculate updraft velocity square(wu2) according to Han et al.'s (2017) \cite han_et_al_2017 equation 7.
+    !LB: This routine outputs updraft velocity square (m/s), updraft omega_u (Pa/s), and cloud average updraft 
+    !velocity (m/s) and omega_u (Pa/s) in the case progsima is true.
+    
+    do k = 1, ktf
+       do i = 1,itf
+          wu2(i,k)=0.
+          drag(i,k)=0.
+          buo(i,k)=0.
+          omega_u(i,k)=0.
+          zeta(i,k)=0.
+          zdqca(i,k)=0.
+       enddo
+    enddo
+
+    do i=1,itf
+       wc(i)=0.
+       omegac(i)=0.
+       sumx(i)=0.
+    enddo
+
+    do k = 1, ktf-1
+       do i = 1,itf
+          zi(i,k) = 0.5*(zo(i,k)+zo(i,k+1))
+          del(i,k) = delp(i,k)*0.001
+       enddo
+    enddo
+    
+    do k = 2, ktf-1
+       do i = 1, itf
+         if (ierr(i)==0) then
+           if(k >= kbcon(i) .and. k < ktcon(i))then
+              gamma   = el2orc * qeso(i,k+1) / (to(i,k+1)**2)
+              if(k >= kbcon(i) .and. clw_all(i,k)>0.)then
+               buo(i,k) = buo(i,k) - g * qlk(i,k)  
+              endif      
+              rfact =  1. + fv * cp * gamma * to(i,k) / xlv
+              buo(i,k) = buo(i,k) + (g / (cp * to(i,k))) * dbyo(i,k) / (1. + gamma) * rfact
+              val = 0.
+              buo(i,k) = buo(i,k) + g * fv * max(val,(qeso(i,k) - qo(i,k)))
+              buo(i,k) = max(val,buo(i,k))
+              drag(i,k) = max(entr_rate_2d(i,k),cd(i,k))
+           endif
+        endif
+       enddo
+    enddo
+
+      bb1 = 4.0
```

Comment:
If these (`bb1, bb2`) are tunable, consider moving their set statements to just after the declaration section and maybe write a comment about their value, range, etc.?